### DISCUSSION TITLE
how to get test Elmessage in Vitest

### DISCUSSION THREAD

**Discussion by wangjunxian** [element-plus/element-plus#13079] [Q&A] [answered]
I'm using Vitest to do test.
and I want to validate the message is right.
I have tried many ways to get Elmessge in my Vitest code.
but still failed.
may anyone tell me how to get Elmessage in Vitest?
thank you.

**Comment by ryuhangyeong** [comment]
Will this help you?

- https://github.com/element-plus/element-plus/blob/dev/packages/components/message/__tests__/message-manager.test.tsx
- https://github.com/element-plus/element-plus/blob/dev/packages/components/message-box/__tests__/message-box.test.ts

**Reply by wangjunxian** [reply]
Thank you for your reply. I have seen this code. But it's so different from what we used.
luckily, I have solved this problem. this is the code how I solved it.
「const successSpy = vi.spyOn(ElMessage, 'success');
expect(successSpy).toHaveBeenCalledWith('handle it')」
but unfortunately, I have many many other problems, first one is about el-table, I don't know how to call function [handleCurrentChange] in Vitest. I have tried many ways for one day. but couldn't find a way to deal with it.
I'm wondering whether you know anything about it.
very sorry for troubling you so many times. and I would like to pay for it 
I really could't catch up with my schedule.
this is my e-mail [wangjunxian2016@gmail.com]
Best regards

**Reply by ryuhangyeong** [reply, marked as answer]
[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtdGFibGVcbiAgICByZWY9XCJzaW5nbGVUYWJsZVJlZlwiXG4gICAgOmRhdGE9XCJ0YWJsZURhdGFcIlxuICAgIGhpZ2hsaWdodC1jdXJyZW50LXJvd1xuICAgIHN0eWxlPVwid2lkdGg6IDEwMCVcIlxuICAgIEBjdXJyZW50LWNoYW5nZT1cImhhbmRsZUN1cnJlbnRDaGFuZ2VcIlxuICA+XG4gICAgPGVsLXRhYmxlLWNvbHVtbiB0eXBlPVwiaW5kZXhcIiB3aWR0aD1cIjUwXCIgLz5cbiAgICA8ZWwtdGFibGUtY29sdW1uIHByb3BlcnR5PVwiZGF0ZVwiIGxhYmVsPVwiRGF0ZVwiIHdpZHRoPVwiMTIwXCIgLz5cbiAgICA8ZWwtdGFibGUtY29sdW1uIHByb3BlcnR5PVwibmFtZVwiIGxhYmVsPVwiTmFtZVwiIHdpZHRoPVwiMTIwXCIgLz5cbiAgICA8ZWwtdGFibGUtY29sdW1uIHByb3BlcnR5PVwiYWRkcmVzc1wiIGxhYmVsPVwiQWRkcmVzc1wiIC8+XG4gIDwvZWwtdGFibGU+XG4gIDxkaXYgc3R5bGU9XCJtYXJnaW4tdG9wOiAyMHB4XCI+XG4gICAgPGVsLWJ1dHRvbiBAY2xpY2s9XCJzZXRDdXJyZW50KHRhYmxlRGF0YVsxXSlcIj5TZWxlY3Qgc2Vjb25kIHJvdzwvZWwtYnV0dG9uPlxuICAgIDxlbC1idXR0b24gQGNsaWNrPVwic2V0Q3VycmVudCgpXCI+Q2xlYXIgc2VsZWN0aW9uPC9lbC1idXR0b24+XG4gIDwvZGl2PlxuPC90ZW1wbGF0ZT5cblxuPHNjcmlwdCBsYW5nPVwidHNcIiBzZXR1cD5cbmltcG9ydCB7IHJlZiB9IGZyb20gJ3Z1ZSdcbmltcG9ydCB7IEVsVGFibGUgfSBmcm9tICdlbGVtZW50LXBsdXMnXG5cbmludGVyZmFjZSBVc2VyIHtcbiAgZGF0ZTogc3RyaW5nXG4gIG5hbWU6IHN0cmluZ1xuICBhZGRyZXNzOiBzdHJpbmdcbn1cblxuY29uc3QgY3VycmVudFJvdyA9IHJlZigpXG5jb25zdCBzaW5nbGVUYWJsZVJlZiA9IHJlZjxJbnN0YW5jZVR5cGU8dHlwZW9mIEVsVGFibGU+PigpXG5cbmNvbnN0IHNldEN1cnJlbnQgPSAocm93PzogVXNlcikgPT4ge1xuICBzaW5nbGVUYWJsZVJlZi52YWx1ZSEuc2V0Q3VycmVudFJvdyhyb3cpXG59XG5jb25zdCBoYW5kbGVDdXJyZW50Q2hhbmdlID0gKHZhbDogVXNlciB8IHVuZGVmaW5lZCkgPT4ge1xuICBjb25zb2xlLmxvZyh2YWwpXG4gIGN1cnJlbnRSb3cudmFsdWUgPSB2YWxcbn1cbmNvbnN0IHRhYmxlRGF0YTogVXNlcltdID0gW1xuICB7XG4gICAgZGF0ZTogJzIwMTYtMDUtMDMnLFxuICAgIG5hbWU6ICdUb20nLFxuICAgIGFkZHJlc3M6ICdOby4gMTg5LCBHcm92ZSBTdCwgTG9zIEFuZ2VsZXMnLFxuICB9LFxuICB7XG4gICAgZGF0ZTogJzIwMTYtMDUtMDInLFxuICAgIG5hbWU6ICdUb20nLFxuICAgIGFkZHJlc3M6ICdOby4gMTg5LCBHcm92ZSBTdCwgTG9zIEFuZ2VsZXMnLFxuICB9LFxuICB7XG4gICAgZGF0ZTogJzIwMTYtMDUtMDQnLFxuICAgIG5hbWU6ICdUb20nLFxuICAgIGFkZHJlc3M6ICdOby4gMTg5LCBHcm92ZSBTdCwgTG9zIEFuZ2VsZXMnLFxuICB9LFxuICB7XG4gICAgZGF0ZTogJzIwMTYtMDUtMDEnLFxuICAgIG5hbWU6ICdUb20nLFxuICAgIGFkZHJlc3M6ICdOby4gMTg5LCBHcm92ZSBTdCwgTG9zIEFuZ2VsZXMnLFxuICB9LFxuXVxuPC9zY3JpcHQ+XG4iLCJpbXBvcnRfbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsIl9vIjp7fX0=)

I'm not very clear on the difficulty you're running into, but why don't you write your tests by referring to how the example above provides?

example above selects a specific row when the "Select second row" button is pressed.

and I think it's also a way to judge whether it's called with vitest's mock or spyon.
